tests/integ/test_state_machine_definition.py

```diff
@@ -531,4 +584,56 @@ def test_retry_state_machine_creation(sfn_client, sfn_role_arn, training_job_par
         role=sfn_role_arn
     )
 
-    workflow_test_suite(sfn_client, workflow, asl_state_machine_definition, None)
\ No newline at end of file
+    workflow_test_suite(sfn_client, workflow, asl_state_machine_definition, None)
+
+
+def test_state_machine_creation_with_retry_in_constructor(sfn_client, sfn_role_arn, training_job_parameters):
```

Comment:
nit: i don't think this constructor scenario needs it's own integ test, but we should have one for retry (or for catch). 
do we already have an existing test we could expand perhaps?

i'd suggest having a test that sets both a `retry` in the constructor as well as another through `add_retry` if possible. Having said that, if you can't think of a way to simplify, having a test is better than not having a test *always*